Source/WebKit/UIProcess/Automation/mac/WebAutomationSessionMac.mm

```diff
@@ -78,6 +80,21 @@
     NSWindow *window = page.platformWindow();
     auto webView = page.cocoaView();
 
+    // +[NSEvent pressedMouseButtons] does not account for the NSEvent objects created through eventSender JS in tests.
+    // As such, that method always returns 0. To fix this, we swizzle out +[NSEvent pressedMouseButtons], keep track of
+    // the mouse button currently being pressed down, and supply the appropriate return value as specified in documentation.
+    auto methodToSwizzle = class_getClassMethod(objc_getMetaClass(NSStringFromClass([NSEvent class]).UTF8String), @selector(pressedMouseButtons));
+    auto originalImplementation = method_setImplementation(methodToSwizzle, imp_implementationWithBlock([&mouseButtonCurrentlyDown = m_mouseButtonCurrentlyDown] {
+        switch (mouseButtonCurrentlyDown) {
+        case MouseButton::Left:  return 1 << 0;
+        case MouseButton::Right: return 1 << 1;
+        default:                 return 0;
```

Comment:
Nit - I think we usually just write the returns on separate lines in WebKit.

Reply:
Fixed in https://github.com/WebKit/WebKit/commit/a3bd40b1edf94033170f604c71655cca9382e49c